shell/platform/android/io/flutter/embedding/android/FlutterView.java

```diff
@@ -1420,6 +1420,20 @@ public void autofill(SparseArray<AutofillValue> values) {
     textInputPlugin.autofill(values);
   }
 
+  // Invoke this from {@code FlutterActivityAndFragmentDelegate#onStart()}
+  /* package */ void onStart() {
+    if (flutterSurfaceView != null) {
+      flutterSurfaceView.setVisibility(View.VISIBLE);
+    }
+  }
+
+  // Invoke this from {@code FlutterActivityAndFragmentDelegate#onStop()}
+  /* package */ void onStop() {
+    if (flutterSurfaceView != null) {
+      flutterSurfaceView.setVisibility(View.GONE);
+    }
+  }
+
```

Comment:
We've tried to resist adding API like this historically because it's likely to get missed when you don't use the `FlutterActivityAndFragmentDelegate` class, IIRC. @xster or @blasten wdyt?

I'm curious about where the view is getting the wrong visibility set and why this helps. I'm also curious if we're getting calls on `View.invalidateDrawable` at some point here that might help us - if we can take advantage of the View's own lifecycle it'd be better.

Reply:
Or alternatively if we're getting calls to `onDraw` or `onWindowVisibilityChanged`/`onAttachedToWindow`.

Reply:
I guess these methods might not work. In this scenario, the onStart and onStop of the Activity are correct, but the visibility of the window/view tree is still visible after onStop.